### ISSUE TITLE
V3 Demo doesn’t work on Mobile Browsers

### ISSUE THREAD

**Issue by selected-pixel-jameson** [webtiming/timingsrc#27]
I was looking at the V3 demo https://timingsrc.readthedocs.io/en/latest/demoes/demo_mediasync.html#demo-mediasync

I tested in Chrome, Brave and Safari on an iPhone 14 and the videos just don’t play.

**Comment by selected-pixel-jameson** [comment]
Looking at this further it seems like the current version demos don’t work on mobile either and basically crash the browser. Am I missing something?

**Comment by ingararntzen** [comment]
Hi, and thanks for reporting.

It seems to works fine on my end. The only issue I could see was that if the page is reloaded during playback, videos wont start playing, as video playback requires on user activity. Pause resume helps in this case.

Make sure to rewind if if the timing object went all the way to 100.

If you still experience problems, please check the console for any  error messages and report back.

Cheers, Ingar

**Comment by selected-pixel-jameson** [comment]
Is this how it’s supposed to work?

**Comment by ingararntzen** [comment]
Hi. I responded on the github issue.

It's supposed to sync the videos.

lør. 18. mar. 2023 kl. 00:08 skrev selected-pixel-jameson <
***@***.***>:

**Comment by selected-pixel-jameson** [comment]
When I pause and play and reset and play I can see the video being updated. But if I just press play the video frame doesn’t change.

**Comment by selected-pixel-jameson** [comment]
I have a video I’ll try and upload it later.

**Comment by selected-pixel-jameson** [comment]
https://user-images.githubusercontent.com/28204537/226101123-d3ce7d09-383e-49c6-b21c-f5db01b1ea19.mp4

The same thing happens with the current version. This video was taken with Safari, but the same thing happens in Chrome and Brave.

**Comment by selected-pixel-jameson** [comment]
Just looking at this page https://timingsrc.readthedocs.io/en/latest/demoes/demo_mediasync.html#demo-mediasync in Desktop Safari the videos do load, but the playback is extremely Laggy and the videos often get out of sync on my Macbook Pro M1.

**Comment by ingararntzen** [comment]
Hi. 

You're right. This is very far from how it is supposed to look.

Unfortunately though, this is not entirely unexpected on ios devices.

https://timingsrc.readthedocs.io/en/latest/sync/mediasync.html#reservations

It could well be that a new version of the browser is triggering some issue with the sync library.

If you share details about architecture, chrome version and possibly outputs from the developer console, we might be able have a look at it.

**Comment by ingararntzen** [comment]
I'm able to reproduce the kind of trouble you are seeing in safari on a macbook. Not as bad as in your video though, but still pretty useless.

Works just fine in chrome on the same macbook.

**Comment by selected-pixel-jameson** [comment]
Yep. I haven’t had any issue with Chrome or Brave. It works very well in those browsers on Desktop.

**Comment by ingararntzen** [comment]
Unfortunately, not quick resolve on this one.

I have posted a warning on the [demo](https://timingsrc.readthedocs.io/en/latest/demoes/demo_mediasync.html) page.

**Comment by selected-pixel-jameson** [comment]
@ingararntzen Is there any work around? I would possibly be able to create a pull request to fix this, but having some insight into what the actual issue is would be helpful. We have a Start-up that is utilizing this script and getting it working on iOS is imperative to our success.

**Comment by Snarkdoof** [comment]
Could you test some of the videos on https://wp4.demos.mediafutures.no/ and
see if they play well? Both "single device" and "multi device" video are
synchronized to a timing object. These work pretty ok on my iPad at least
even if the demo page you mentioned is rather useless...

N

**Comment by selected-pixel-jameson** [comment]
We are trying to sync at most 4 video elements. Unfortunately our use case requires that the sync be exact so we cannot relax it. Jameson W ParkerOn Mar 28, 2023, at 7:20 AM, Njaal Borch ***@***.***> wrote:﻿
I made a small test page with it, and it might be a problem on iOS that
there are two video elements that it tries to synchronize at the same time.
Is this the use case you are looking for?

It's also possible to add options like {target:0.05} after the timing
object, like mediaSync(element, timingObject, {target:0.05}) to relax the
sync requirement on Safari to be within 50ms of the timingObject.

**Comment by selected-pixel-jameson** [comment]
Any idea on if this can be resolved? 

@Snarkdoof I'm a little confused with what you are saying? If there is only 1 video element on the page how would it be synchronizing the video? Isn't the whole point of the library to be able to have multiple video elements on a page running in sync?

**Comment by Snarkdoof** [comment]
The point is to synchronise a video element to a timing object. Of course,
synchronising multiple video elements is one use case (and one that
normally works well), but it can also be synchronised to other data, sound
tracks etc. Not all devices can play multiple videos at a time (depending
on hw decoders for example), but I can't actually remember if this has been
tested on iOS previously.

We also use multi device services from Motion Corporation to synchronise
video playback om multiple devices, so that's also a good use case.

N

**Comment by selected-pixel-jameson** [comment]
Ok. We have this issue on just Safari as well. It's not limited to Mobile Safari. Our platforms intention is to allow end-user to record music together through the use of Video Elements. 

The gist of it is a user can click record and up to three other video elements will begin playing back in sync while the user is recording. We've seen this work really well with Chrome and Brave, but in Safari the browser just lags. 

 It would be really awesome to figure out if we can or ever will be able to get this working with Safari.

**Comment by nahbee10** [comment]
Hi! I'm having the same issue on all of my ios browsers(safari / chrome / firefox / brave), any chance of work-around or solution to the issue was found? Really hoping to use this for multi channel video installation soon...!! Thank you for sharing this work and maintaining this.

**Comment by Snarkdoof** [comment]
On iOS, all browsers are Safari. It's not allowed with other HTML engines.
I am working on a slightly different solution, but I'm still unsure if this
is solvable on Safari.

N

**Comment by nahbee10** [comment]
@Snarkdoof Thanks for the response! I'll then check if android based browsers are working with the library :)
